architecture/src/iosMain/kotlin/navigation/Navigator.kt

```diff
@@ -352,3 +352,7 @@ class ViewControllerNavigator<A : NavigationAction<*>>(
 
     private fun assertParent() = parent.get().also { assert(it != null) }
 }
+
+// Seems to be a bug on commonizer using new targets sourcesets which is preventing
+// to infer the correct type as it was doing before
```

Comment:
Commonizer was not used before, as there was no `iosMain` sourceset so there is no "before"